mmcv/device/ipu/dataloader.py

```diff
@@ -97,17 +100,17 @@ class IPUDataLoader(poptorch.DataLoader):
     """
 
     def __init__(self,
-                 dataset,
-                 options,
-                 batch_size=1,
-                 shuffle=False,
-                 num_workers=0,
-                 drop_last=True,
-                 persistent_workers=True,
-                 auto_distributed_partitioning=True,
-                 mode='sync',
-                 async_options=None,
-                 rebatched_worker_size=None,
+                 dataset: torch.utils.data.Dataset,
+                 options: poptorch.Options,
+                 batch_size: Optional[int] = 1,
+                 shuffle: Optional[bool] = False,
+                 num_workers: Optional[int] = 0,
+                 drop_last: Optional[bool] = True,
+                 persistent_workers: Optional[bool] = True,
+                 auto_distributed_partitioning: Optional[bool] = True,
+                 mode: Optional[poptorch.DataLoaderMode] = 'sync',
+                 async_options: Optional[Dict[str, Any]] = None,
+                 rebatched_worker_size: Optional[int] = None,
```

Comment:
Only if the default value is None, the `Optional` could be used. Suggest simply remove the `Optional`.

Reply:
done